viewing_party/party.py

```diff
@@ -1,23 +1,116 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+
+    if title and genre and rating:
+        movie_dict['title'] = title
+        movie_dict['genre'] = genre
+        movie_dict['rating'] = rating
+
+        return movie_dict
+    
+    return None
+def add_to_watched(user_data, movie):
+
+    user_data['watched'].append(movie)
+    return user_data
+def add_to_watchlist(user_data, movie):
+
+    user_data['watchlist'].append(movie)
+    return user_data
+def watch_movie(user_data, title):
+
+    for movie_dict in user_data['watchlist']:
+        if title == movie_dict['title']:
+            user_data['watched'].append(movie_dict)
+            user_data['watchlist'].remove(movie_dict)
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
-
-
+def get_watched_avg_rating(user_data):
+    total_rating = 0.0
+    avg_rating = 0.0
+    for movie_dict in user_data['watched']:
+        total_rating += movie_dict['rating']
+        if len(user_data['watched']) == 0:

```

Comment:
Recall that `0` in Python is falsey, so you could write 
```python
if not len(user_data['watched']):
```
instead of comparing it to `0`.